### PR TITLE
chore: Separate workflow to add/remove 'stale' label

### PR DESCRIPTION
**Pull request by totoprayogo1916**
**Description**
Follow-up https://github.com/codeigniter4/CodeIgniter4/pull/7957

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: .github/workflows/label-add-conflict-all-pr.yml [hunk 1]
```diff
@@ -1,14 +1,10 @@
-name: Auto Add Label "stale" & Comment Conflicts
+name: Auto Add Label "stale" & Comment Conflicts for All PR
```
Comment (kenjis) [inline .github/workflows/label-add-conflict-all-pr.yml:1]:
```suggestion
name: Auto Label "stale" for All PRs
```

File: .github/workflows/label-add-remove-conflict-single-pr.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: Auto Add/Remove "stale" label in single PR
```
Comment (kenjis) [inline .github/workflows/label-add-remove-conflict-single-pr.yml:1]:
```suggestion
name: Auto Label "stale"
```

File: .github/workflows/label-add-conflict-all-pr.yml [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
     branches:
       - develop
       - '4.*'
-  pull_request:
-    branches:
-      - develop
-      - '4.*'
 
 jobs:
   auto-label-comment-conflict:
```
Comment (kenjis) [inline .github/workflows/label-add-conflict-all-pr.yml:10]:
```suggestion
  build:
    name: Check Conflicts
```

File: .github/workflows/label-add-remove-conflict-single-pr.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - '4.*'
+
+jobs:
+  check-conflict:
```
Comment (kenjis) [inline .github/workflows/label-add-remove-conflict-single-pr.yml:10]:
```suggestion
  build:
    name: Check Conflicts
```

**Review by kenjis — APPROVED** [review]

File: .github/workflows/label-add-conflict-all-pr.yml [hunk 5]
```diff
@@ -1,17 +1,12 @@
...
-  push:
-    branches:
-      - develop
-      - '4.*'
-  pull_request:
-    branches:
-      - develop
-      - '4.*'
+  schedule:
+    - cron: '0 */6 * * *' # Run at every 6 hours
```
[the remark this answers is not in the document]
Comment (kenjis) [reply .github/workflows/label-add-conflict-all-pr.yml:5]: Checking mergeability takes time. So if we get UNKNOWN, we may need to wait and get the PR again.
However, this check does not have to be done every time, so it is fine to make it a cronjob.

> The value of the mergeable attribute can be true, false, or null. If the value is null, then GitHub has started a background job to compute the mergeability. After giving the job time to complete, resubmit the request. When the job finishes, you will see a non-null value for the mergeable attribute in the response. If mergeable is true, then merge_commit_sha will be the SHA of the test merge commit.
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

[the remark this answers is not in the document]
Comment (kenjis) [reply .github/workflows/label-add-conflict-all-pr.yml:5]: @totoprayogo1916 For the moment, would you like this PR to be merged for testing?
Or do you want to check more of this workflow in your repository?

**Review by kenjis — APPROVED** [review]
Not sure if it would work without the cronjob. 
But let's merge and see.

**Comment by kenjis** [comment]
#7986 has conflict, but not labeled.

**Comment by kenjis** [comment]
> When https://github.com/codeigniter4/CodeIgniter4/pull/7986 conflicts, is there a push on the develop or 4.x branch?

Yes. #7967 was merged.
And the action is: https://github.com/codeigniter4/CodeIgniter4/actions/runs/6332904494/job/17200180659

**Comment by kenjis** [comment]
Yes.

And now I got my PRs (#7955, #7981) to be labeled after merging #7990

**Comment by kenjis** [comment]
> The value of the mergeable attribute can be true, false, or null. **If the value is null, then GitHub has started a background job to compute the mergeability. After giving the job time to complete, resubmit the request.** When the job finishes, you will see a non-null value for the mergeable attribute in the response. If mergeable is true, then merge_commit_sha will be the SHA of the test merge commit.
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request

**Comment by ddevsr** [comment]
I testing with schedule as already suggested by @totoprayogo1916, always revalidate limitable latest PR by get latest status mergeable when schedule running.

https://github.com/PHPDevsr/check-conflict-action/actions
https://github.com/PHPDevsr/playground/actions

**Comment by kenjis** [comment]
I think "Run at every 6 minutes" is running too often.

In the case of UNKNOWN, I think if we wait some seconds before retrieving again, we can get the result.

**Comment by kenjis** [comment]
Check Conflicts Remove label failed.
See #7992

**Comment by ddevsr** [comment]
Before conflict https://github.com/codeigniter4/CodeIgniter4/actions/runs/6309422686/job/17129306013 success running with read permission because no have label stale to remove using write permission. And after conflict bot trigger added label and comment using. Many action running using permission read, I think this section of permission job.

> https://github.com/PHPDevsr/check-conflict-action/actions
> https://github.com/PHPDevsr/playground/actions

You can see 2 action running well with write permission

**Comment by kenjis** [comment]
@ddevsr Sorry, I don't get what you say.
If you can fix the issue, please send a PR.

**Comment by kenjis** [comment]
Removed remove label-add-remove-conflict-single-pr.yml, because it does not work at all. 
See #7999

**Pull request merged**